src/main/java/edu/upf/taln/dri/lib/model/util/ObjectGenerator.java

```diff
@@ -1600,307 +1599,17 @@ else if(titleLang.trim().toLowerCase().equals("es")) {
 				// ******************************************************
 				// List of authors with name, surname, email, affiliation
 				String sourceDocFeature = GateUtil.getStringFeature(cacheManager.getGateDoc(), "source").orElse(null);
-				if(sourceDocFeature != null && sourceDocFeature.equals(SourceENUM.GROBID.toString())) {
```

Comment:
The if-statement from GROBID based on `sourceDocFeature` got extracted into `GROBIDAuthorExtractor` which is picked by the `ExtractorFactory`.

We leave it to `ObjectGenerator` to convert `sourceDocFeature` into the enum. If it fails (or the extractors fail) an exception is caught here and should continue normal execution without author extraction.